Register and document rbac.authorization.kubernetes.io/autoupdate annotation (#36162)
kubernetes/website: content/en/docs/reference/labels-annotations-taints/_index.md

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.

### rbac.authorization.kubernetes.io/autoupdate

> Examples: `rbac.authorization.kubernetes.io/autoupdate: false`

Reviewer: ```suggestion
Example: `rbac.authorization.kubernetes.io/autoupdate: "false"`
```
Annotation values are strings.

Author: Done.

> Used on: ClusterRole, ClusterRoleBinding

Reviewer 2: and Role/RoleBinding

Author: Done.

> When this annotation is set to `true`, default RBAC ClusterRole and ClusterRoleBinding objects are automatically updated at server start to add missing permissions and subjects (extra permissions and subjects are left in place). To prevent autoupdating a particular role or rolebinding, set this annotation to `false`.

Reviewer: If I make my own ClusterRole and set this annotation, what happens?

Reviewer 2: `kubectl auth reconcile` (which allows reconciling arbitrary RBAC objects in a manifest file) will respect the annotation

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.

### rbac.authorization.kubernetes.io/autoupdate

Example: `rbac.authorization.kubernetes.io/autoupdate: "false"`

Used on: ClusterRole, ClusterRoleBinding, Role, RoleBinding

When this annotation is set to `true`, default RBAC ClusterRole and ClusterRoleBinding objects are automatically updated at server start to add missing permissions and subjects (extra permissions and subjects are left in place). To prevent autoupdating a particular role or rolebinding, set this annotation to `false`.

> If user create their own ClusterRole, `kubectl auth reconcile` (which allows reconciling arbitrary RBAC objects in a manifest file) will respect this annotation.

Reviewer: (nit)
```suggestion
If you create your own ClusterRole and set this annotation, `kubectl auth reconcile`
(which allows reconciling arbitrary RBAC objects in a {{< glossary_tooltip text="manifest" term_id="manifest" >}}) respects this annotation and does not automatically add missing permissions and
subjects.
```
Is this right?

See [Enforcing Pod Security at the Namespace Level](/docs/concepts/security/pod-security-admission)
for more information.

### rbac.authorization.kubernetes.io/autoupdate

Example: `rbac.authorization.kubernetes.io/autoupdate: "false"`

Used on: ClusterRole, ClusterRoleBinding, Role, RoleBinding

> When this annotation is set to `true`, default RBAC ClusterRole and ClusterRoleBinding objects are automatically updated at server start to add missing permissions and subjects (extra permissions and subjects are left in place). To prevent autoupdating a particular role or rolebinding, set this annotation to `false`.

Reviewer 2: ```suggestion
When this annotation is set to `true` on default RBAC objects created by the kube-apiserver, they are automatically updated at server start to add missing permissions and subjects (extra permissions and subjects are left in place). To prevent autoupdating a particular role or rolebinding, set this annotation to `false`.
```

Reviewer: See https://github.com/kubernetes/website/pull/36162#discussion_r1072301369 for a discussion about the annotation values, which are actually strings.

Reviewer: (I didn't spot this before)

> If you create your own ClusterRole and set this annotation, `kubectl auth reconcile`

Reviewer 2: ```suggestion
If you create your own RBAC objects and set this annotation to `false`, `kubectl auth reconcile`
```

Reviewer: (nit) we'd typically prefer to write ”false” without backticks - see https://kubernetes.io/docs/contribute/style/style-guide/#use-normal-style-for-string-and-integer-field-values

However, to make it clear that this is actually the **string** `"false"`, try:
```suggestion
If you create your own RBAC objects and set this annotation to `"false"`, `kubectl auth reconcile`
```
